clusterloader2/docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,303 @@
+    - basename: test-deployment
+      objectTemplatePath: "deployment.yaml"
+      templateFillMap:
+        Replicas: 10
+- name: Wait for pods to be running
+  measurements:
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: gather
+- name: Measure pod startup latency
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: gather
+```
+
+By default, clusterloader will delete auto-created namespaces so we don't need to worry with cleaning up cluster.
+
+Now, in order to finish our first test, we need to specify deployment template.
+You can think of it as regular kubernetes object, but with templating.
+Clusterloader by default adds parameter `Name` that you can use in your template.
+In our config, we also passed `Replicas` parameter.
+We need to remember to set correct labels so PodStartupLatency and WaitForControlledPodsRunning will watch correct pods.
+So our template for deployment will look like this (`deployment.yaml` file):
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Name}}
+  labels:
+    group: test-pod
+spec:
+  replicas: {{.Replicas}}
+  selector:
+    matchLabels:
+      group: test-pod
+  template:
+    metadata:
+      labels:
+        group: test-pod
+    spec:
+      containers:
+      - image: k8s.gcr.io/pause:3.1
+        imagePullPolicy: IfNotPresent
+        name: {{.Name}}
+        resources:
+          requests:
+            cpu: 1m
+            memory: 10M
+```
+## Execute test
+Before running test, make sure that kubeconfig current context points to kind cluster:
+```bash
+kubectl config current-context
+kind-test-cluster
+```
+To execute test, run:
+```bash
+cd clusterloader2/
+go run cmd/clusterloader.go --testconfig=config.yaml --provider=kind --kubeconfig=${HOME}/.kube/config --v=2
+```
+
+At the end of clusterloader output you should see pod startup latency:
+```json
+{
+  "data": {
+    "Perc50": 7100.534796,
+    "Perc90": 8702.523037,
+    "Perc99": 9122.894555
+  },
+  "unit": "ms",
+  "labels": {
+    "Metric": "pod_startup"
+  }
+},
+```
+`pod_startup` measures time since pod was created until it was observed via watch as running.
+
+You should also see that test succeeded:
+```
+--------------------------------------------------------------------------------
+Test Finished
+Test: ./config.yaml
+Status: Success
+--------------------------------------------------------------------------------
+```
+As an exercies you can modify threshold for PodStartupLatency below values you've observed in your run and check if test fails.
+## Delete kind cluster
+To delete kind cluster, run:
+```bash
+kind delete cluster --name test-cluster
+```
+
+## Running 100-node scale test
+
+Here you can find general purpose [Load test]. This test is release-blocking test we use to evaluate scalability of kubernetes.
+It consists of 3 main phases:
+- Creating objects
+- Scaling objects in range (-50%, 50%)
```

Comment:
Maybe "Randomly scaling objects to the size between 50 % and 150 % of their original sizes"?